### PR TITLE
Add sqlite

### PR DESCRIPTION
**Pull request by dcostea**
I had to add an Id property for the Metric entity, as well. I need it for SQLite db. I hope is not a problem.

### CODE REVIEW

File: MLOps.NET.SQLite/Entities/Experiment.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using MLOps.NET.Entities.Entities;
+using System;
+
+namespace MLOps.NET.SQLite.Entities
+{
+    public sealed class Experiment : IExperiment
```
Comment (aslotte) [inline MLOps.NET.SQLite/Entities/Experiment.cs:6]: We can probably make this internal

File: MLOps.NET.SQLite/Entities/Metric.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿using MLOps.NET.Entities.Entities;
+using System;
+
+namespace MLOps.NET.SQLite.Entities
+{
+    public sealed class Metric : IMetric
```
Comment (aslotte) [inline MLOps.NET.SQLite/Entities/Metric.cs:6]: Same here in regards to internal

File: MLOps.NET.SQLite/Entities/Run.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+﻿using MLOps.NET.Entities.Entities;
+using System;
+
+namespace MLOps.NET.SQLite.Entities
+{
+    public sealed class Run : IRun
```
Comment (aslotte) [inline MLOps.NET.SQLite/Entities/Run.cs:6]: Same here in regards to internal

File: MLOps.NET.SQLite/Storage/LocalDbContext.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.EntityFrameworkCore;
+using MLOps.NET.SQLite.Entities;
+
+namespace MLOps.NET.SQLite.Storage
+{
+    public class LocalDbContext : DbContext
```
Comment (aslotte) [inline MLOps.NET.SQLite/Storage/LocalDbContext.cs:6]: I'm not a SQLite expert, but would it make sense not to limit this to Local databases? E.g. I would image an end user probably wanting to have a central place for storage that may not be the same resource as the model is trained on.

File: MLOps.NET.SQLite/Storage/SQLiteMetaDataStore.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+using System;
+using System.Threading.Tasks;
+
+namespace MLOps.NET.Storage
+{
+    public sealed class SQLiteMetaDataStore : IMetaDataStore
+    {
+        public async Task<Guid> CreateExperimentAsync(string name)
+        {
+
```
Comment (aslotte) [inline MLOps.NET.SQLite/Storage/SQLiteMetaDataStore.cs:12]: Extra newline

File: MLOps.NET.SQLite/Storage/SQLiteMetaDataStore.cs [hunk 6]
```diff
@@ -0,0 +1,47 @@
+﻿using MLOps.NET.SQLite.Entities;
+using MLOps.NET.SQLite.Storage;
+using System;
+using System.Threading.Tasks;
+
+namespace MLOps.NET.Storage
+{
+    public sealed class SQLiteMetaDataStore : IMetaDataStore
```
Comment (aslotte) [inline MLOps.NET.SQLite/Storage/SQLiteMetaDataStore.cs:8]: Could this be made internal?

File: MLOps.NET.SQLite/Storage/LocalDbContext.cs (hunk 4 above)
Comment (dcostea) [reply MLOps.NET.SQLite/Storage/LocalDbContext.cs:6]: SQLite is file-based whereas MSSQL Server and MySQL are server-based. I think it is not meant for concurrence, therefore it is better to be local.
For the client-server scenario, we can use MSSQL, but in storage project.

Comment (aslotte) [reply MLOps.NET.SQLite/Storage/LocalDbContext.cs:6]: Gotcha, that was the missing piece of the puzzle, thanks!

**Comment by aslotte** [comment]
@dcostea I couldn't see an extension method to instantiate the SQLiteMetaDataStore. You may want to add one, e.g. `UseSQLite()`.

Would you be able to test it real quick locally as well to make sure it appears to work? If not, I'm happy to try it during one of the streams. Thanks!

**Comment by dcostea** [comment]
> @dcostea I couldn't see an extension method to instantiate the SQLiteMetaDataStore. You may want to add one, e.g. `UseSQLite()`.
> 
> Would you be able to test it real quick locally as well to make sure it appears to work? If not, I'm happy to try it during one of the streams. Thanks!

I have added UseSQLite. I will try to test later, otherwise, we can do it in the next stream.

**Pull request merged**